### ISSUE TITLE
Clang 4 + libc++ builds failing on Travis

### ISSUE THREAD

**Issue by CaseyCarter** [ericniebler/range-v3#516]
...reliably since [2016-11-10](https://travis-ci.org/ericniebler/range-v3/builds/174868115):
```
cd /home/travis/build/ericniebler/range-v3/build/test && /usr/bin/clang++-4.0    -I/home/travis/build/ericniebler/range-v3/include  -fsanitize=address,undefined,integer -fno-omit-frame-pointer -fno-sanitize=unsigned-integer-overflow -D__extern_always_inline=inline -stdlib=libc++ -I/usr/include/c++/v1/ -std=c++11 -ftemplate-backtrace-limit=0 -Weverything -Werror -pedantic-errors -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-weak-vtables -Wno-padded -Wno-undef -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-shadow -Wno-old-style-cast -Wno-documentation -Wno-documentation-unknown-command -march=native -mtune=native -O0 -fno-inline -g3 -fstack-protector-all   -o CMakeFiles/istream_range.dir/istream_range.cpp.o -c /home/travis/build/ericniebler/range-v3/test/istream_range.cpp
In file included from /home/travis/build/ericniebler/range-v3/test/istream_range.cpp:1:
In file included from /home/travis/build/ericniebler/range-v3/include/range/v3/istream_range.hpp:17:
In file included from /usr/include/c++/v1/istream:163:
In file included from /usr/include/c++/v1/ostream:138:
In file included from /usr/include/c++/v1/ios:216:
In file included from /usr/include/c++/v1/__locale:15:
In file included from /usr/include/c++/v1/string:468:
In file included from /usr/include/c++/v1/string_view:166:
In file included from /usr/include/c++/v1/__string:55:
In file included from /usr/include/c++/v1/algorithm:636:
In file included from /usr/include/c++/v1/initializer_list:47:
/usr/include/c++/v1/cstddef:43:15: fatal error: 'stddef.h' file not found
#include_next <stddef.h>
              ^~~~~~~~~~
1 error generated.
make[2]: *** [test/CMakeFiles/istream_range.dir/istream_range.cpp.o] Error 1
make[2]: Leaving directory `/home/travis/build/ericniebler/range-v3/build'
make[1]: *** [test/CMakeFiles/istream_range.dir/all] Error 2
make[1]: Leaving directory `/home/travis/build/ericniebler/range-v3/build'
make: *** [all] Error 2
```

Clang4 + libstdc++ succeeds. Last successful build was on 2016-11-02, presumably breakage is due to a libc++ change that occurred sometime between those two dates.

**Comment by CaseyCarter** [comment]
Paging range-v3 clang + libc++ build gurus. @gnzlbg @EricWF

**Comment by CaseyCarter** [comment]
Closed by #520.
